addons/l10n_my_edi/models/account_move.py

```diff
@@ -0,0 +1,661 @@
+
+                error, status_fetch_result = self._l10n_my_get_submission_status(submission_uid, company.l10n_my_edi_proxy_user_id)
+                if error:
+                    raise UserError(error)  # We do not expect errors here so raising is a correct solution.
+                for invoice in invoices:
+                    invoice_result = status_fetch_result.get(invoice.l10n_my_edi_external_uuid)
+                    new_state = invoice_result['status']
+                    if new_state == invoice.l10n_my_edi_state:
+                        continue
+
+                    # If the state changed, we update the invoice with the new state and an eventual reason.
+                    invoice._l10n_my_edi_set_status(
+                        state=new_state,
+                        message=_('This invoice has been %(status)s for reason: %(reason)s', status=new_state, reason=invoice_result['reason']) if invoice_result.get('reason') else None,
+                    )
+                submission_processed += 1
+                # Commit if we can, in case an issue arises later.
+                if self._can_commit():
+                    self._cr.commit()
+        if total_submissions_to_process > submission_processed:
+            self.env.ref('l10n_my_edi.ir_cron_myinvois_sync')._trigger()
+
+    @api.model
+    def _l10n_my_get_submission_status(self, submission_uid, proxy_user):
+        """ Returns the status of all invoices in the submission.
+        If there are too many and the submission is paginated, this will fetch each page with a waiting time of 3s per call.
+        As a page can hold 100 invoice, it should not happen often.
+
+        The proxy user is given as a param as this method can be called from the cron, in which case we can't rely on self.
+        """
+        # In case of errors, we return it alongside any results. We cannot raise as this is called from the send & print in some cases.
+        error = ''
+        # The api returns the result per document uuid, already correctly formated.
+        # We do not need to format the data anymore for processing later but just to ensure we get complete data of the submission.
+        result = proxy_user._l10n_my_edi_contact_proxy(
+            endpoint='api/l10n_my_edi/1/get_submission_statuses',
+            params={
+                'submission_uid': submission_uid,
+                'page': 1,
+            }
+        )
+
+        if 'error' in result:
+            error = self._l10n_my_edi_map_error(result['error'])
+        else:
+            if result['document_count'] <= 100:  # If so, we got all of it at once.
+                result = result['statuses']
+            else:
+                # Otherwise we'll need to get the remaining invoices per batch of 100.
+                for page in range(2, (result['document_count'] // 100) + 1):
+                    time.sleep(1)  # To avoid any risks of throteling, we should wait a bit before continuing
```

Comment:
1 second or 3 seconds?

Reply:
Sorry, 1s indeed.
The API asks to not repeat the call for the same submission more than once every 3-5s
Here we are getting different submissions and considering there will be a delay while waiting for their response/... 1s seems enough.